test/nectar_test.exs

```diff
@@ -2,11 +2,30 @@ defmodule NectarTest do
   use ExUnit.Case
   doctest Nectar
 
-  test "a known good HTTP server" do
-    assert {:ok, %HTTPoison.Response{}} = HTTPoison.get("http://elixir-lang.org/")
+  # https://en.wikipedia.org/wiki/Ephemeral_port#Range
+  def start_server, do: start_server(49152)
```

Comment:
Ahh, that's a clearer way to show that.

1.) The wikipedia page said.

> Many Linux kernels use the port range 32768 to 61000

That you can't access > 61000 via `/proc` I'm not sure if that is a problem on some systems? That's why I went with a lower range.

2.) I think it can take a bit of time for it to fail to open the port when it is in use. I'm not sure if it is a good idea to try all of them (it might block for a _long_ time). Also, you would probably have better luck picking random ones, in case something is using a range of ports. Maybe?

Reply:
Yeah, that will work. There is still a chance that it could pick wrong many times and hang for a while but it's pretty unlikely.